src/common-imports.ts

```diff
@@ -0,0 +1,32 @@
+import Vue, {VueConstructor} from 'vue'
+
+import * as commonComponents from '@/components/common'
+import * as interactionDesignerComponents from '@/components/interaction-designer'
+import * as blockEditorsComponents from '@/components/interaction-designer/block-editors'
+import * as blockTypesComponents from '@/components/interaction-designer/block-types'
+import * as flowEditorsComponents from '@/components/interaction-designer/flow-editors'
+import * as flowImportComponents from '@/components/interaction-designer/flow-editors/import'
+import * as toolbarComponents from '@/components/interaction-designer/toolbar'
+import * as blocksComponents from '@/components/interaction-designer/blocks'
+import * as blockResourceEditorsComponents from '@/components/interaction-designer/resource-editors'
+import * as resourceEditorComponents from '@/components/resource-editor'
+
+require('./font-awesome-icon')
+
+export default function registerCustomComponents(extra = {}) {
+  Object.entries({
+    ...commonComponents,
+    ...interactionDesignerComponents,
+    ...blockEditorsComponents,
+    ...blockTypesComponents,
+    ...flowEditorsComponents,
+    ...flowImportComponents,
+    ...toolbarComponents,
+    ...blocksComponents,
+    ...blockResourceEditorsComponents,
+    ...resourceEditorComponents,
+    ...extra,
+  }).forEach(([name, component]: [string, any]) => {
```

Comment:
What would be the correct way to reference a Vue component here instead of `any`? I'm confused because there are both classic style components and class-based ones in the map.

Reply:
I'm not sure there is a good solution here without spending maybe too long thinking about it... Given this is what we had before I'm okay with this solution for now.